searchlib/src/vespa/searchlib/queryeval/filter_wrapper.h

```diff
@@ -0,0 +1,63 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "searchiterator.h"
+#include "blueprint.h"
+#include <vespa/searchlib/fef/termfieldmatchdata.h>
+#include <vespa/searchlib/fef/termfieldmatchdataarray.h>
+
+namespace search::queryeval {
+
+/**
+ * Wraps an iterator for use as a filter search.
+ * Owns TermFieldMatchData the wrapped iterator
+ * can wire to, and write to if necessary.
+ **/
+class FilterWrapper : public SearchIterator {
+private:
+    std::vector<fef::TermFieldMatchData> _unused_md;
+    fef::TermFieldMatchDataArray _tfmda;
+    std::unique_ptr<SearchIterator> _wrapped_search;
+public:
+    FilterWrapper(const Blueprint::State &state)
```

Comment:
prefer sending the number of fields directly instead of the state to reduces unneeded coupling